googletest/include/gtest/gtest.h

```diff
@@ -658,6 +661,13 @@ class GTEST_API_ TestResult {
   // Increments the death test count, returning the new count.
   int increment_death_test_count() { return ++death_test_count_; }
 
+  // Increments the successful assert count, returning the new count.
+  size_t increment_success_assert_count(size_t v = 1) { return success_assert_count_ += v; }
+
+  // Set/get executed flag.
+  bool was_executed() const { return executed_; }
```

Comment:
This data member is redundant, the value can be calculated run-time between Passed(), "Failed" , "Skipped". Please remove

Reply:
As far as I recall the problem is that Skipped will only count tests that were actually attempted executed. If let's say you have an environment issue which causes the tests never to be even considered, they are not counted as skipped. We had runs where no tests were being run because conditions checked by AddGlobalTestEnvironment callbacks caused them to not be attempted run at all.

Reply:
Do we have an example of a Skipped test? There's none in the "golden" output?

Reply:
There's also no JSON or XML output of DISABLED tests, which is of the same category as FILTERED tests, which is the natural pattern to follow IMHO.